### PR TITLE
Fix the Debian Package for SharedFx, Publish all to our Repo

### PR DESCRIPTION
**Pull request by brthor**
cc @livarcocc @anurse 

Let's see if this works

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/1890)

### CODE REVIEW

**Comment by livarcocc** [comment]
:shipit:

**Comment by livarcocc** [comment]
Still failing. Also, squash.

**Comment by piotrpMSFT** [comment]
Ok, we're getting closer.

It looks like we no longer have `corehost` in the package. This, in general, is fine but needs to be accounted for in the tests. Our test infrastructure was executing the built binaries by passing them to `corehost` which no longer exists. I updated the infrastructure to use `dotnet` instead. This gets us closer, but seems to have exposed a bug in dotnet itself:

````
23:10:18     Microsoft.DotNet.Tests.EndToEnd.EndToEndTest.TestDotnetBuild IL]
23:10:18       Command did not output with Expected Output. Expected: Hello World!
23:10:18       
23:10:18       File Name: /opt/code/artifacts/ubuntu.14.04-x64/stage2/bin/dotnet
23:10:18       Arguments: "/opt/code/artifacts/ubuntu.14.04-x64/tests/artifacts/f9198682-801b-4671-8b08-99f9c465d324/e2etestroot/test space/bin/e2etestroot.dll"
23:10:18       Exit Code: 1
23:10:18       StdOut:
23:10:18       No executable found matching command "dotnet-/opt/code/artifacts/ubuntu.14.04-x64/tests/artifacts/f9198682-801b-4671-8b08-99f9c465d324/e2etestroot/test space/bin/e2etestroot.dll"
23:10:18       
````

It looks like `dotnet` is not recognizing "/opt/code/artifacts/ubuntu.14.04-x64/tests/artifacts/f9198682-801b-4671-8b08-99f9c465d324/e2etestroot/test space/bin/e2etestroot.dll" as the path to an executable and is instead handing it off to dotnet.dll. dotnet.dll then fails to recognize that string as a built-in command and tries to find a packaged command. That then fails due to that command not existing.

@schellap, is this a known issue for dotnet.exe? why is it not recognizing and executing the assembly?

**Comment by schellap** [comment]
Wait, you have dotnet.dll next to dotnet.exe, then you are a standalone app.

**Comment by schellap** [comment]
This brings an interesting point,

The CLI tests like e2etestroot either should be a standalone app or be a portable app. Right now they are some weird app which get activated using dotnet.exe from the CLI bin drop instead of the muxer. If you used corehost.exe instead of dotnet.exe it would work. But this is not a supported activation model.

**Comment by schellap** [comment]
@gkhanna79 stuff we talked about yesterday...

**Comment by anurse** [comment]
> It looks like we no longer have corehost in the package.

Why did that change? We need to continue shipping corehost as `corehost` for now because of the limitations in the host as mentioned by @schellap. As long as it is named `dotnet`, it can't distinguish between being used as a standalone app or being used to launch a specific app.

**Comment by anurse** [comment]
I definitely still see `corehost` in the `stage2` directory, unless the debian package is removing it somehow...

**Comment by schellap** [comment]
Using corehost is still a hack to get it working as is. The tests should be fixed to be portable or standalone and not frankenapp.

**Comment by anurse** [comment]
The test apps are always either portable or standalone. The problem is we're trying to launch them with the `dotnet` executable in the SDK, which is immediately adjacent to `dotnet.dll` which causes `corehost` to behave in CoreConsole mode.

**Comment by anurse** [comment]
I think I know the issue. `corehost` is in the CLI package, but not on the `PATH`, so we need the full path. I can get that. I'll update the PR and see if that resolves the issue.

File: packaging/debian/dotnet-nightly-debian_config.json [hunk 1]
```diff
@@ -28,9 +28,8 @@
     },
 
     "debian_dependencies":{
-        "libssl1.0.0" : {},
         "clang-3.5" : {},
-        "libcurl3" : {}
```
Comment (brthor) [inline packaging/debian/dotnet-nightly-debian_config.json]: this should stay

Comment (brthor) [inline packaging/debian/dotnet-nightly-debian_config.json]: ah but it doesn't need to be in the sdk

**Comment by naamunds** [comment]
@brthor Will this PR re-enable the publishing of Debian packages?  It doesn't look like it re-enables the `PublishDebFileToDebianRepo` target in PublishTargets.cs which was disabled in https://github.com/dotnet/cli/pull/1947.  Is the publishing going to be done a different way now, or is the change to re-enable publishing going to come later?

**Comment by naamunds** [comment]
(cc: @dleeapho @MichaelSimons)

File: scripts/dotnet-cli-build/PublishTargets.cs [hunk 2]
```diff
@@ -47,7 +47,7 @@ public static BuildTargetResult Publish(BuildTargetContext c)
         [Target(
             nameof(PublishTargets.PublishVersionBadge),
             nameof(PublishTargets.PublishSdkInstallerFile),
-            //nameof(PublishTargets.PublishDebFileToDebianRepo),
+            nameof(PublishTargets.PublishDebFilesToDebianRepo),
```
Comment (brthor) [inline scripts/dotnet-cli-build/PublishTargets.cs]: @naamunds

**Comment by brthor** [comment]
@dotnet-bot test this please

File: scripts/dotnet-cli-build/PublishTargets.cs (hunk 2 above)
Comment (naamunds) [inline scripts/dotnet-cli-build/PublishTargets.cs]: Can we uncomment `BuildPlatform.Ubuntu` for `PublishSdkInstallerFile`?  I see that `PublishDebFilesToDebianRepo` is no longer dependent on `PublishSdkInstallerFile`, so perhaps it's not necessary yet.

File: scripts/dotnet-cli-build/PublishTargets.cs [hunk 3]
```diff
@@ -110,6 +120,38 @@ public static BuildTargetResult PublishDebFileToDebianRepo(BuildTargetContext c)
...
+            var uploadUrl = $"https://dotnetcli.blob.core.windows.net/dotnet/{Channel}/Installers/{Version}/{Path.GetFileName(installerFile)}";
+            var uploadJson = GenerateUploadJsonFile(packageName, Version, uploadUrl);
+
+            Cmd(Path.Combine(Dirs.RepoRoot, "scripts", "publish", "repoapi_client.sh"), "-addpkg", uploadJson)
+                    .Execute()
+                    .EnsureSuccessful();
+
+            return c.Success();
+        }
+
```
Comment (naamunds) [inline scripts/dotnet-cli-build/PublishTargets.cs]: Consider refactoring the common logic in `PublishSdkDebToDebianRepo`, `PublishSharedFrameworkDebToDebianRepo`, and `PublishSharedHostDebToDebianRepo` into a private method that those three would each call.

File: scripts/dotnet-cli-build/PublishTargets.cs (hunk 2 above)
Comment (brthor) [inline scripts/dotnet-cli-build/PublishTargets.cs]: cc @Sridhar-MS do you know why this was commented out? didn't even notice until now.

File: scripts/dotnet-cli-build/PublishTargets.cs (hunk 3 above)
Comment (brthor) [inline scripts/dotnet-cli-build/PublishTargets.cs]: I agree this whole file could stand for some factoring in a separate PR. :smile: 

Intention here is to quickly bring up the scenario, we have a number of places where code improvements are needed but I'd prefer to have something working to iterate on.

Comment (brthor) [inline scripts/dotnet-cli-build/PublishTargets.cs]: For such obvious changes as this I'd usually put in a quick cleanup commit, but with the CI times we are seeing and our deb packages on the floor I'd like to get this in first.

**Comment by brthor** [comment]
We're green on Ubuntu! Just waiting on 2016 to get this in.

File: scripts/dotnet-cli-build/PublishTargets.cs (hunk 2 above)
Comment (brthor) [inline scripts/dotnet-cli-build/PublishTargets.cs]: looked more into this, I don't think this function should get run on ubuntu, but its name is somewhat misleading. I will file and take an issue to clean up this file.

**Pull request merged**